packages/core/src/hub.ts

```diff
@@ -532,6 +549,30 @@ export function getCurrentHub(): Hub {
   return getHubFromCarrier(registry);
 }
 
+/**
+ * Sets the global async context strategy
+ */
+export function setAsyncContextStrategy(strategy: AsyncContextStrategy): void {
+  // Get main carrier (global for every environment)
+  const registry = getMainCarrier();
+  registry.__SENTRY__ = registry.__SENTRY__ || {};
+  registry.__SENTRY__.asyncContextStrategy = strategy;
+}
+
+/**
+ * Runs the given callback function with the global async context strategy
```

Comment:
let's add warning to the docstrings that these are private APIs and have no semvar guarantees

Reply:
I can use `@ignore`?

Is this actually private though? Wont users be able to use this themselves for async context isolation?

Reply:
For now we make it private, we can remove this label later on.